src/modules/jsUtils.js

```diff
@@ -0,0 +1,28 @@
+function compareObjects(o1, o2) {
+    // If either are undefined, return false - don't consider undefined to equal undefined.
+    if (typeof o1 === 'undefined' || typeof o1 === 'undefined') {
+        return false;
+    }
+
+    if (Array.isArray(o1)) {
+        if (!Array.isArray(o2)) {
+            return false;
+        }
+        // Simple array comparison - expects same order and only scalar values
+        return o1.length === o2.length && o1.every(function (v, i) {
```

Comment:
Do we care about the scenario when the order of elements is different? For our scenario, are two arrays with the same elements in different order not equal?